### ISSUE TITLE
support .apl files

### ISSUE THREAD

**Issue by colin-combe** [Rappsilber-Laboratory/xiSPEC_ms_parser#11]
related https://github.com/Rappsilber-Laboratory/XiSearch/issues/22

**Comment by colin-combe** [comment]
motivated by fact xiSearch reads .apl, but converting files before search might be better

**Comment by lutzfischer** [comment]
we have various implementation for converting apl to mgf - problem is the how unknown-charge states are handled. potentially breaking the index.

maxquant writes spectra with unknown charge state o *.peka.apl - but twice - ones as doubly charged and ones as triply charged. and I am not sure if the order is consistent (especially if you use the allspectra*.apl files)
Xi just ignores the the triply charged entries of the .peak.apl and consideres the doubly charged as unknown chargestates. 

If you want to support apl - then i need to ensure that the index is correct for the spectrum when returned from xi

**Comment by colin-combe** [comment]
> we have various implementation for converting apl to mgf - problem is the how unknown-charge states are handled. potentially breaking the index.

I don't think I follow - i'm thinking if everything's converted to .mgf *before* its fed to xiSearch then the indexes should be correct when it comes out the other end?

> If you want to support apl 

I don't really want to support another non-standard format, I'd prefer we recommended converting .apl files sooner rather than later if that's a workable solution.

**Comment by lutzfischer** [comment]
>> we have various implementation for converting apl to mgf - problem is the how unknown-charge states are handled. potentially breaking the index.
> 
> I don't think I follow - i'm thinking if everything's converted to .mgf before its fed to xiSearch then the indexes should be correct when it comes out the other end?

Yes if converted before search then everything should be fine. But there is the alternative approach to convert the files after search - maybe because you did not convert it and the search did take a month, then researching becomes an effort. This still works fairly straight forward - as long as no _peak_ spectra (unknown charge state - actually a lot more is comparatively vague about these) where searched. 
Convert each file on its own and rename in the xi-result everything from .apl to .mgf.

> I don't really want to support another non-standard format, I'd prefer we recommended converting .apl files sooner rather than later if that's a workable solution.

I agree that it would be better if we could ignore apl. It's a dependency on a format that is only internally used by maxquant and could therefore change or disappear in any future version of maxquant.

Also it comes per default with a lot less information then we get usually from mgf-files.

But it seems there are still cases where apl-files give still better results then our mgf-pipeline...

**Comment by colin-combe** [comment]
Also there's no support for .apl in pyteomics (which the parser uses) or any other open source lib I've found. I can't even really find documentation for the format. Another downside of proprietary formats.
